File: react/src/lib/domain/entities/TypeProjectElement.ts
import BaseEnum from './BaseEnum';

export default class TypeProjectElement extends BaseEnum {
    public static SOURCE_CODE = new TypeProjectElement('1', 'project:projectElementsEnum:generatedCode');
    public static MAPS = new TypeProjectElement('2', 'project:projectElementsEnum:maps');
    public static TABLE_DATA_DICTIONARY = new TypeProjectElement('3', 'project:projectElementsEnum:tableDD');
    public static BACKUPS = new TypeProjectElement('4', 'project:projectElementsEnum:backups');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}

File: backend/src/main/java/com/equisoft/dca/backend/project/model/TypeProjectElement.java
package com.equisoft.dca.backend.project.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum TypeProjectElement implements BaseEnum<Integer> {
	SOURCE_CODE(1, "Source code"),
	MAPS(2, "Maps"),
	TABLE_DATA_DICTIONARY(3, "Table data dictionary"),
	BACKUPS(4, "Backups");

	private final int id;

	private final String name;

	TypeProjectElement(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}
